File: Auxiliary/gclib/examples/vb/examples/examples/Position_Tracking_Example.vb
Module Position_Tracking_Example
    Function Main() As Integer
        Dim rc As Integer = Examples.GALIL_EXAMPLE_OK
        Dim gclib As Gclib = New Gclib()
        Dim args() As String = Environment.GetCommandLineArgs()
        Dim speed As Integer = 0
        Try
            If args.Count = 2 Then
                speed = 5000
            ElseIf args.Count = 3 Then

                Dim ok As Boolean = Byte.TryParse(args(2), speed)

                If Not ok Then
                    Console.WriteLine("An invalid speed was entered.  " +
                                      "Please enter a valid integer for speed.")
                    Console.WriteLine("Usage: position_tracking_example.exe <ADDRESS> <SPEED=5000>")

                    Console.Write(vbCrLf + "Press any key to close the example.")
                    Console.ReadKey()
                    Return Examples.GALIL_EXAMPLE_ERROR
                End If

            Else
                Console.WriteLine("Incorrect number of arguments provided")
                Console.WriteLine("Usage: position_tracking_example.exe <ADDRESS> <SPEED=5000>")
				
                Console.Write(vbCrLf + "Press any key to close the example.")
                Console.ReadKey()
                Return Examples.GALIL_EXAMPLE_ERROR
            End If


            Dim address As String = args(1) 'Retrieve address from command line
            gclib.GOpen(address) 'Open a connection at the provided address

            rc = Examples.Position_Tracking(gclib, speed) 'Begin position tracking mode
        Catch ex As Exception
            Examples.PrintError(gclib, ex)
            rc = Examples.GALIL_EXAMPLE_ERROR
        Finally
            gclib.GClose()
        End Try

        Console.Write(vbCrLf + "Press any key to close the example.")
        Console.ReadKey()

        Return rc
    End Function
End Module

File: Auxiliary/gclib/examples/cs/examples/examples/position_tracking_example.cs
/** @addtogroup cs_examples
  * @{
  */
  
/*! \file position_tracking_example.cs
*
* See \link examples.Examples.Position_Tracking() Position_Tracking() \endlink for implementation of logic
* <br><br>For VB.NET, see definition in file position_tracking_example.vb
*/
using System;
using System.Linq;

namespace examples
{
    /** @addtogroup cs_examples
    * @{
    */
	/// <summary>
	/// Places controller into position tracking mode.  Accepts user-defined positional values 
	/// at the command line.
	/// </summary>
	/// <remarks>The first argument should be the IP Address of a Galil controller.\n
	///	The second argument is optional and defines the default speed of the controller in
	/// Position Tracking mode.</remarks>
	/*! For VB.NET, see definition in file position_tracking_example.vb */
    class Position_Tracking_Example
    {
        /// <summary>
        /// Main function for the position tracking example.
        /// </summary>
        /// <param name="args">An array of command line arguments.</param>
        /// <returns>The success status or error code of the function.</returns>
        /// <remarks>The first argument should be the IP Address of a Galil controller.
		///	The second argument is optional and defines the default speed of the controller in
		/// Position Tracking mode.</remarks>
        public static int Main(string[] args)
        {
            int rc = Examples.GALIL_EXAMPLE_OK;
            gclib gclib = new gclib();

            try
            {
                int speed = 0;
                
                if(args.Count() == 1)
                {
                    speed = 5000;
                }
                else if(args.Count() == 2)
                {
                    bool ok = int.TryParse(args[1], out speed);

                    if(!ok)
                    {
                        Console.WriteLine("An invalid speed was entered.  " +
                                          "Please enter a valid integer for speed.");
                        Console.WriteLine("Usage: position_tracking_example.exe <ADDRESS> <SPEED=5000>");

                        Console.Write("\nPress any key to close the example");
                        Console.ReadKey();
                        return Examples.GALIL_EXAMPLE_ERROR;
                    }
                }
                else
                {
                    Console.WriteLine("Incorrect number of arguments provided");
                    Console.WriteLine("Usage: position_tracking_example.exe <ADDRESS> <SPEED=5000>");
					
                    Console.Write("\nPress any key to close the example");
                    Console.ReadKey();					
                    return Examples.GALIL_EXAMPLE_ERROR;
                }

                string address = args[0]; //Retrieve address from command line
                gclib.GOpen(address); // Open a connection at the provided address

                rc = Examples.Position_Tracking(gclib, speed); // Begin position tracking mode
            }
            catch(Exception ex)
            {
                Examples.PrintError(gclib, ex);
                rc = Examples.GALIL_EXAMPLE_ERROR;
            }
            finally
            {
                gclib.GClose();
            }

            Console.Write("\nPress any key to close the example");
            Console.ReadKey();

            return rc;
        }
    }
/** @}*/
}
/** @}*/